File: leetcode/1608-special-array-with-x-elements-greater-than-or-equal-x.py
from collections import defaultdict


class Solution:
    def specialArray(self, nums: List[int]) -> int:
        freq = defaultdict(int)
        n = len(nums)

        for num in nums:
            freq[min(num, n)] += 1

        greater_equal_than_count = 0
        for i in range(n, 0, -1):
            greater_equal_than_count += freq[i]
            if greater_equal_than_count == i:
                return i
        return -1

File: src/1608-special-array-with-x-elements-greater-than-or-equal-x.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var specialArray = function (nums) {
    let l = nums.length
    //     -1
    while (l >= 0) {
        //    2
        let total = 0
        for (let i = 0; i <= nums.length - 1; i++) {
            if (nums[i] >= l) total++
        }
        if (total === l) {
            return total
        } else {
            l--
            total = 0
        }
    }
    return -1
};